app/factories/bulkrax/valkyrie_object_factory.rb

```diff
@@ -252,7 +252,7 @@ def self.ordered_file_sets_for(object)
 
     def delete(user)
       obj = find
-      return false unless obj
+      raise ObjectFactoryInterface::ObjectNotFoundError, "Object not found to delete" unless obj
```

Comment:
It is misleading for a deletion to not happen but be reported as deleted. This ensures that the one running the importer knows about the failure.